### DISCUSSION TITLE
How to fit MS+compact binary using phoebe?

### DISCUSSION THREAD

**Discussion by dingxu6207** [phoebe-project/phoebe2#842] [Questions]
I want to fit the MS+compact binary in detached modle using phoebe.
How to set the parameters of the compact star using Phoebe?

**Comment by Raindogjones** [comment]
Currently, the only option for fitting compact stars in phoebe is to use blackbody atmospheres and manual limb-darkening. In the near future, we hope to include support for hot, compact stars through the incorporation of TMAP atmospheres but for now they have to be modelled as blackbodies.

**Reply by dingxu6207** [reply]
I have set the atmospheres.
b.set_value_all('ld_mode_bol','manual')
b.set_value_all('ld_mode','manual')
b.set_value_all('atm','blackbody')

and set 
b.set_value('distortion_method', component='secondary', value='none')
b.run_compute(model='distortion_method_none')

The parameters of the primary star need to be adjusted.
The secondary is the compact star. Do the parameters of the compact star, such as Teff and radius, need to be adjusted?

**Reply by Raindogjones** [reply]
Depending on the parameters of your system, you might be able to keep ck2004 or phoenix atmospheres for the main sequence primary.

As for the compact secondary, whether or not you need to change its parameters will depend on what it is and what impact it has on the primary.  If it contributes directly to the light curve and/or irradiates the primary, then you will need to adjust the temperature and radius.

**Reply by dingxu6207** [reply]
Thank you. 
b.set_value('distortion_method', component='secondary', value='none')  value = 'none ' means no flux contribution.
In this condition,  do the parameters of the compact star, such as Teff and radius, need to be adjusted?

**Reply by Raindogjones** [reply]
No. With distortion_method='none' no mesh is created for the star, so the Teff and radius are irrelevant.
http://phoebe-project.org/docs/2.4/examples/distortion_method_none

**Reply by dingxu6207** [reply]
b.set_value(qualifier='atm', component='secondary', value='blackbody')
b.set_value(qualifier='ld_mode', component='secondary', value='manual')

There is a error ValueError: 2 results found: ['ld_mode@secondary@lc01@lc@dataset', 'ld_mode@secondary@rv01@rv@dataset'] 

How to set the manual limb-darkening for the compact star

**Reply by kecnry** [reply]
You can either set the value for the lc and rv datasets independently (by passing `dataset='lc01'`, etc) or set both by using `set_value_all` instead.

**Reply by dingxu6207** [reply]
Many Thanks!

**Reply by dingxu6207** [reply]
![微信截图_20240404232245](https://github.com/phoebe-project/phoebe2/assets/32772701/f5d27309-6363-410d-b26c-c602c9dc6a0c)


Is there any introduction about the eclipse method ‘only horizon’ in Phoebe?  I can't find it.

**Reply by kecnry** [reply]
It seems that is missing from http://phoebe-project.org/docs/2.4/tutorials/eclipse (I have added a note to add it in the future).  In short, this will handle the horizon (only including elements facing the observer) but will ignore eclipses between stars themselves.

**Reply by dingxu6207** [reply]
Thank you. In MS+WD binary systems, when only one radial velocity curve (Primary star) is observed and the light curve shows ellipsoidal variations, it can be understood that the ellipsoidal variations are generated by the main sequence (MS) star, not by eclipses. Can this model (the eclipse method ‘only horizon’ ) be used for analysis?

**Reply by kecnry** [reply]
Yes, although it never hurts to compare the actual model to ensure eclipses shouldn't be present/significant.

**Reply by dingxu6207** [reply]
Many thanks!
